File: src/main/scala/info/crlog/higgs/protocols/boson/BosonType.scala
package info.crlog.higgs.protocols.boson

/**
 * The set of data types supported by the Boson protocol
 * The value (int) of each type is the flag used to prefix the payload of the given type
 * @author Courtney Robinson <courtney@crlog.info>
 */
object BosonType {
  //val converter = new ByteConverter()
  //base data types
  val BYTE = 1
  val SHORT = 2
  val INT = 3
  val LONG = 4
  val FLOAT = 5
  val DOUBLE = 6
  val BOOLEAN = 7
  val CHAR = 8
  val NULL = 9
  val STRING = 10
  val ARRAY = 11
  val LIST = 12
  val MAP = 13
  val POLO = 14

  //request response flags
  val REQUEST_METHOD_NAME = -127
  val REQUEST_PARAMETERS = -126
  val REQUEST_CALLBACK = -125
  val RESPONSE_METHOD_NAME = -124
  val RESPONSE_PARAMETERS = -123
}

File: boson/src/main/java/io/higgs/boson/BosonType.java
package io.higgs.boson;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class BosonType {
    protected BosonType() {
    }

    public static final int BYTE = 1;
    public static final int SHORT = 2;
    public static final int INT = 3;
    public static final int LONG = 4;
    public static final int FLOAT = 5;
    public static final int DOUBLE = 6;
    public static final int BOOLEAN = 7;
    public static final int CHAR = 8;
    public static final int NULL = 9;
    public static final int STRING = 10;
    public static final int ARRAY = 11;
    public static final int LIST = 12;
    public static final int MAP = 13;
    public static final int POLO = 14;
    public static final int REFERENCE = 15;
    public static final int SET = 16;
    public static final int ENUM = 17;
    //request response flags
    public static final int REQUEST_METHOD_NAME = -127;
    public static final int REQUEST_PARAMETERS = -126;
    public static final int REQUEST_CALLBACK = -125;
    public static final int RESPONSE_METHOD_NAME = -124;
    public static final int RESPONSE_PARAMETERS = -123;
}
